src/map/npc.c

```diff
@@ -4777,32 +4777,38 @@ static const char *npc_parse_mapflag(const char *w1, const char *w2, const char
 	char mapname[32];
 	int state = 1;
 
-	nullpo_retr(strchr(start,'\n'), w1);
-	nullpo_retr(strchr(start,'\n'), w3);
+	nullpo_retr(strchr(start, '\n'), w1);
+	nullpo_retr(strchr(start, '\n'), w3);
 
 	// w1=<mapname>
-	if( sscanf(w1, "%31[^,]", mapname) != 1 )
+	if (sscanf(w1, "%31[^,]", mapname) != 1)
 	{
-		ShowError("npc_parse_mapflag: Invalid mapflag definition in file '%s', line '%d'.\n * w1=%s\n * w2=%s\n * w3=%s\n * w4=%s\n", filepath, strline(buffer,start-buffer), w1, w2, w3, w4);
-		if (retval) *retval = EXIT_FAILURE;
-		return strchr(start,'\n');// skip and continue
+		ShowError("npc_parse_mapflag: Invalid mapflag definition in file '%s', line '%d'.\n * w1=%s\n * w2=%s\n * w3=%s\n * w4=%s\n", filepath, strline(buffer, start - buffer), w1, w2, w3, w4);
+		if (retval) 
+			*retval = EXIT_FAILURE;
+		return strchr(start, '\n');// skip and continue
 	}
 	m = map->mapname2mapid(mapname);
 	if (m < 0) {
 		ShowWarning("npc_parse_mapflag: Unknown map in file '%s', line '%d': %s\n * w1=%s\n * w2=%s\n * w3=%s\n * w4=%s\n",
-		            filepath, strline(buffer,start-buffer), mapname, w1, w2, w3, w4);
-		if (retval) *retval = EXIT_FAILURE;
-		return strchr(start,'\n');// skip and continue
+			filepath, strline(buffer, start - buffer), mapname, w1, w2, w3, w4);
+		if (retval) 
+			*retval = EXIT_FAILURE;
+		return strchr(start, '\n');// skip and continue
 	}
 
-	if (w4 && !strcmpi(w4, "off"))
+	if (w4 && strcmpi(w4, "off") == 0)
 		state = 0; //Disable mapflag rather than enable it. [Skotlex]
 
-	if (!strcmpi(w3, "nosave")) {
+	if (strcmpi(w3, "nomemo") == 0)
+		map->list[m].flag.nomemo = (state) ? 1 : 0;
```

Comment:
```suggestion
		map->list[m].flag.nomemo = (state != 0) ? 1 : 0;
```
Same for the other changes in file.

Reply:
updated